src/tools/ceph_objectstore_tool.cc

```diff
@@ -861,14 +861,24 @@ int get_attrs(
 	  cout << "snapset " << snapset << std::endl;
 	  if (!snapset.is_legacy()) {
 	    for (auto& p : snapset.clone_snaps) {
-	      hobject_t clone = hoid.hobj;
-	      clone.snap = p.first;
+	      ghobject_t clone = hoid;
+	      clone.hobj.snap = p.first;
 	      set<snapid_t> snaps(p.second.begin(), p.second.end());
+	      if (!store->exists(coll, clone)) {
```

Comment:
You can remove last_head too.  Just replace "hold == last_head" test with hold.is_head().